qiskit/chemistry/ground_state_calculation/mes_ground_state_calculation.py

```diff
@@ -0,0 +1,178 @@
+
+"""Ground state computation using a minimum eigensolver."""
+
+from typing import Union, List, Any, Optional, Dict
+
+import numpy as np
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Instruction
+from qiskit.quantum_info import Statevector
+from qiskit.result import Result
+from qiskit.aqua.algorithms import MinimumEigensolver
+from qiskit.aqua.operators import OperatorBase, WeightedPauliOperator, StateFn, CircuitSampler
+from qiskit.chemistry import FermionicOperator
+from qiskit.chemistry.drivers import BaseDriver
+from qiskit.chemistry.ground_state_calculation import GroundStateCalculation
+from qiskit.chemistry.qubit_transformations import QubitOperatorTransformation
+from qiskit.chemistry.results import EigenstateResult
+
+from .mes_factories import MESFactory
+
+
+class MinimumEigensolverGroundStateCalculation(GroundStateCalculation):
+    """Ground state computation using a minimum eigensolver."""
+
+    def __init__(self, transformation: QubitOperatorTransformation,
+                 solver: Union[MinimumEigensolver, MESFactory]) -> None:
+        """
+
+        Args:
+            transformation: Qubit Operator Transformation
+            solver: Minimum Eigensolver or MESFactory object, e.g. the VQEUCCSDFactory.
+        """
+        super().__init__(transformation)
+        self._solver = solver
+
+    @property
+    def solver(self) -> Union[MinimumEigensolver, MESFactory]:
+        """Returns the minimum eigensolver or factory."""
+        return self._solver
+
+    @solver.setter
+    def solver(self, solver: Union[MinimumEigensolver, MESFactory]) -> None:
+        """Sets the minimum eigensolver or factory."""
+        self._solver = solver
+
+    def returns_groundstate(self) -> bool:
+        """TODO
+        whether the eigensolver returns the ground state or only ground state energy."""
+
+        return False
+
+    def compute_groundstate(self, driver: BaseDriver,
+                            aux_operators: Optional[List[Any]] = None
+                            ) -> EigenstateResult:
+        """Compute Ground State properties.
+
+        Args:
+            driver: a chemistry driver object which defines the chemical problem that is to be
+                    solved by this calculation.
+            aux_operators: Additional auxiliary operators to evaluate at the ground state.
+                Depending on whether a fermionic or bosonic system is solved, the type of the
+                operators must be ``FermionicOperator`` or ``BosonicOperator``, respectively.
+
+        Raises:
+            NotImplementedError: If an operator in ``aux_operators`` is not of type
+                ``FermionicOperator``.
+
+        Returns:
+            An eigenstate result. Depending on the transformation this can be an electronic
+            structure or bosonic result.
+        """
+        if aux_operators is not None:
+            if any(not isinstance(op, (WeightedPauliOperator, FermionicOperator))
```

Comment:
How/where is FermionicOperator (or BosonicOperator) being transformed a qubit operator? WeightedPauliOp is a qubit operator. The min eig solver only takes qubit operators (as type LegcacyBaseOperator or OperatorBase)